agent/whatweb_agent.py

```diff
@@ -81,38 +81,97 @@ def process(self, message: msg.Message) -> None:
+                continue
+            else:
+                with tempfile.NamedTemporaryFile() as fp:
+                    self._start_scan(target.name, fp.name)
+                    self._parse_emit_result(
+                        target.name, fp, int(target.port), target.schema)
 
-    def _prepare_target(self, message: msg.Message) -> Target:
-        """Returns a target object to be scanned."""
+    def _prepare_targets(self, message: msg.Message) -> List[Target]:
+        """Returns a list of target objects to be scanned."""
+        targets = []
+        domain_targets = self._prepare_domain_targets(message)
+        ip_targets = self._prepare_ip_targets(message)
+        targets.extend(domain_targets)
+        targets.extend(ip_targets)
+        return targets
+
+    def _prepare_domain_targets(self, message: msg.Message) -> List[Target]:
+        """Returns a list of domain targets to be scanned."""
+        targets = []
         if message.data.get('url') is not None:
             target = self._get_target_from_url(message.data['url'])
+            targets.append(target)
         elif message.data.get('name') is not None:
             domain_name = message.data['name']
-            target = Target(domain=domain_name, schema=self.args.get('schema'), port=self.args.get('port'))
+
+            if message.data.get('port') is not None:
+                port = message.data['port']
+            else:
+                port = self.args.get('port')
+            if message.data.get('schema') is not None:
+                schema = message.data['schema']
+            else:
+                schema = self.args.get('schema')
+            target = Target(name=domain_name, schema=schema, port=port)
+            targets.append(target)
+
+        return targets
+
+    def _prepare_ip_targets(self, message: msg.Message) -> List[Target]:
+        """Returns a list of ip targets to be scanned."""
+        targets = []
+        host = message.data.get('host')
+        mask = message.data.get('mask')
+
+        if host is None:
+            return targets
+        if message.data.get('port') is not None:
+            port = message.data['port']
         else:
-            raise NotImplementedError(f'Message selector {message.selector} not supported.')
+            port = self.args.get('port')
+        if message.data.get('protocol') is not None:
+            schema = message.data['protocol']
+        else:
+            schema = self.args.get('schema')
 
-        return target
+        if mask is not None:
+            try:
+                addresses = ipaddress.ip_network(f'{host}/{mask}')
+                for address in addresses.hosts():
+                    targets.append(Target(name=str(address), schema=schema, port=port))
+            except ValueError as e:
+                logger.error('Invalid IP or mask. %s', e)
+        else:
+            try:
+                addresses = ipaddress.ip_network(host)
+                for address in addresses.hosts():
```

Comment:
no need for the for loop here since it is a single address, you can just do return str(ipaddress.ip_network(host))